### ISSUE TITLE
Problems Parsing string in Xamarin Android

### ISSUE THREAD

**Issue by versacex** [toddams/RazorLight#141]
After following your example I found that the Executing Assembly needed to be set as well in Xamarin for Android.  However after passing this issue I came accross another exception which is internal          
            
```
string razorText = @"User id is @Model.UserID";
RazorLightEngineBuilder builder = new RazorLightEngineBuilder();
builder.SetOperatingAssembly(Assembly.GetExecutingAssembly());
RazorLightEngine engine = builder.Build();

string output = await engine.CompileRenderAsync("templateKey", razorText, new SettingsViewModel() { UserID = 357 });
```

This is the codeblock which errors out with a null reference exception with no further information on the compile line.  After pressing continue I get another error:

01-12 04:04:44.721 I/MonoDroid( 7234): System.NullReferenceException: Object reference not set to an instance of an object.
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.DefaultMetadataReferenceManager.Resolve (Microsoft.Extensions.DependencyModel.DependencyContext dependencyContext) [0x0000b] in <79984452719b4e08baa554379632421b>:0 
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.DefaultMetadataReferenceManager.Resolve (System.Reflection.Assembly assembly) [0x00007] in <79984452719b4e08baa554379632421b>:0 
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.RoslynCompilationService.EnsureOptions () [0x0003b] in <79984452719b4e08baa554379632421b>:0 
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.RoslynCompilationService.get_ParseOptions () [0x00000] in <79984452719b4e08baa554379632421b>:0 
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.RoslynCompilationService.CreateSyntaxTree (Microsoft.CodeAnalysis.Text.SourceText sourceText) [0x00000] in <79984452719b4e08baa554379632421b>:0 
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.RoslynCompilationService.CreateCompilation (System.String compilationContent, System.String assemblyName) [0x0000d] in <79984452719b4e08baa554379632421b>:0 
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.RoslynCompilationService.CompileAndEmit (RazorLight.Generation.IGeneratedRazorTemplate razorTemplate) [0x0001b] in <79984452719b4e08baa554379632421b>:0 
01-12 04:04:44.721 I/MonoDroid( 7234):   at RazorLight.Compilation.RoslynCompilationService.CompileAsync (RazorLight.Generation.IGeneratedRazorTemplate razorTemplate) [0x00093] in <79984452719b4e08baa554379632421b>:0

**Comment by toddams** [comment]
Since Xamarin is running on mono - this issue is related https://github.com/dotnet/cli/issues/3219

Looks like we need to find another way of fetching project dependencies, or add a fallback mechanism

**Comment by toddams** [comment]
Fixed in #159
